### PR TITLE
PRECOMIT: base_address_parish

### PR DESCRIPTION
**Pull request by mamfredym**

### CODE REVIEW

File: base_address_parish/views/res_parish_view.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
```
Comment (celm1990) [inline base_address_parish/views/res_parish_view.xml]: `data` is not need, can be removed

File: base_address_parish/views/res_partner_view.xml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      <field name="name">parish.partner.view</field>
+      <field name="model">res.partner</field>
+      <field name="inherit_id" ref="base.view_partner_form" />
+      <field name="arch" type="xml">
+        <field name="city" position="attributes">
+          <attribute name="invisible">1</attribute>          
+        </field>
+        <field name="city" position="after">
+          <field name="city_id" placeholder="City" class="o_address_city" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
+        </field>
```
Comment (celm1990) [inline base_address_parish/views/res_partner_view.xml]: Because this??

Please see https://github.com/odoo/odoo/blob/15.0/addons/base_address_city/models/res_partner.py#L42-L60

File: base_address_parish/views/res_partner_view.xml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      <field name="name">parish.res.partner.form.address</field>
+      <field name="model">res.partner</field>
+      <field name="inherit_id" ref="base.view_partner_address_form" />
+      <field name="arch" type="xml">
+        <field name="city" position="attributes">
+          <attribute name="invisible">1</attribute>          
+        </field>
+        <field name="city" position="after">
+          <field name="city_id" placeholder="City" class="o_address_city" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
+        </field>
```
Comment (celm1990) [inline base_address_parish/views/res_partner_view.xml]: same here

File: base_address_parish/views/res_partner_view.xml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      <field name="name">parish.res.partner.form.private</field>
+      <field name="model">res.partner</field>
+      <field name="inherit_id" ref="base.res_partner_view_form_private" />
+      <field name="arch" type="xml">
+          <field name="city" position="attributes">
+            <attribute name="invisible">1</attribute>          
+          </field>
+          <field name="city" position="after">
+            <field name="city_id" placeholder="City" class="o_address_city" attrs="{'readonly': [('type', '=', 'contact'),('parent_id', '!=', False)]}"/>
+          </field>
```
Comment (celm1990) [inline base_address_parish/views/res_partner_view.xml]: same here

File: .gitignore [hunk 5]
```diff
@@ -6,6 +6,7 @@ __pycache__/
 
 # C extensions
 *.so
+*.conf
```
Comment (cormaza) [inline .gitignore:9]: You shouldn't change this file

File: base_address_parish/README.rst [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+This module is part of the `OCA/partner-contact <https://github.com/OCA/partner-contact/tree/15.0/partner_industry_secondary>`_ project on GitHub.
+
+You are welcome to contribute. To learn how please visit https://odoo-community.org/page/Contribute.
```
Comment (cormaza) [inline base_address_parish/README.rst:93]: This file is generated by oca-gen-addon-readme, you may asure that there's files called CONTRIBUTRORS.rst and DESCRIPTION.rst on folder readme inside module, you can check here

https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: base_address_parish/__manifest__.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2015 Antiun Ingenieria S.L. - Javier Iniesta
```
Comment (cormaza) [inline base_address_parish/__manifest__.py]: Sure this is the copyrigth?

File: base_address_parish/i18n/base_address_parish.po [hunk 8]
```diff
@@ -0,0 +1,83 @@
+# Translation of Odoo Server.
```
Comment (cormaza) [inline base_address_parish/i18n/base_address_parish.po]: You should generate this file like .pot file, then oca bot generate per project Webtable space for transalate

https://translation.odoo-community.org/

File: base_address_parish/i18n/es.po [hunk 9]
```diff
@@ -0,0 +1,83 @@
+# Translation of Odoo Server.
```
Comment (cormaza) [inline base_address_parish/i18n/es.po:1]: this file should be deleteted

File: base_address_parish/i18n/es_EC.po [hunk 10]
```diff
@@ -0,0 +1,83 @@
+# Translation of Odoo Server.
```
Comment (cormaza) [inline base_address_parish/i18n/es_EC.po:1]: this file should be deleted

File: base_address_parish/models/res_parish.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+class ResParish(models.Model):
+    _name = "res.parish"
+    _order = "name"
+
+    name = fields.Char("Name", required=True, translate=True)
+    code = fields.Char("Code")
+    country_id = fields.Many2one('res.country', string='Country', required=True)
+    state_id = fields.Many2one(
+        'res.country.state', 'State', domain="[('country_id', '=', country_id)]")
+    city_id = fields.Many2one(
```
Comment (cormaza) [inline base_address_parish/models/res_parish.py:13]: If parish can belong to city, why you should specific country or state?, this fields(country_id, state_id) can be related with res.city fields, with this design you can set different country for a city, or state

File: base_address_parish/models/res_parish.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+        res = []
+        for parish in self:
+            name = parish.name if not parish.code else '%s (%s)' % (parish.name, parish.code)
+            res.append((parish.id, name))
+        return res
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get_uid=None):
+        args = list(args or [])
+        # optimize out the default criterion of ``ilike ''`` that matches everything
```
Comment (cormaza) [inline base_address_parish/models/res_parish.py]: remove comments

File: base_address_parish/models/res_partner.py [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+class Partner(models.Model):
+    _inherit = 'res.partner'
+
+    country_enforce_parishes = fields.Boolean(related='country_id.enforce_parishes', readonly=True)
+    parish_id = fields.Many2one('res.parish', string='Parish')
+
+    @api.onchange('parish_id')
+    def _onchange_parish_id(self):
+        if self.parish_id:
+            self.zip = self.parish_id.code
```
Comment (cormaza) [inline base_address_parish/models/res_partner.py]: If code of parish is the zip on partner, field should has same name, I'm not sure if this implementation is rigth because zip code can be different on the same parish, please share us the base for this design

File: base_address_parish/views/res_country_view.xml [hunk 14]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data>
+    <record id="view_res_country_parish_extended_form" model="ir.ui.view">
+        <field name="model">res.country</field>
+        <field name="inherit_id" ref="base.view_country_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//div[hasclass('oe_button_box')]" position="inside">
+                <button name="%(action_res_parish_tree)d"
```
Comment (cormaza) [inline base_address_parish/views/res_country_view.xml]: Parishes belongs to city, I think that you can add this action on res.city, and maybe add a specific menu to edit all parish records

**Review by cormaza — CHANGES_REQUESTED** [review]
Please check review, thanks for contribution

File: base_address_parish/views/res_partner_view.xml (hunk 2 above)
Comment (mamfredym) [reply base_address_parish/views/res_partner_view.xml]: Because in base_address_city don't have view for show city_id, so, I show beacuse parish it's depending of city_id

File: base_address_parish/views/res_parish_view.xml (hunk 1 above)
Comment (mamfredym) [reply base_address_parish/views/res_parish_view.xml]: Ok, thanks

File: base_address_parish/__manifest__.py (hunk 7 above)
Comment (mamfredym) [reply base_address_parish/__manifest__.py]: Thanks

File: base_address_parish/views/res_partner_view.xml (hunk 2 above)
Comment (celm1990) [reply base_address_parish/views/res_partner_view.xml]: > Because in base_address_city don't have view for show city_id, so, I show beacuse parish it's depending of city_id

Of course, fields are added to form view from python code, see link

Comment (celm1990) [reply base_address_parish/views/res_partner_view.xml]: see also https://github.com/odoo/odoo/blob/15.0/odoo/addons/base/models/res_partner.py#L49-L67

File: base_address_parish/models/res_partner.py (hunk 13 above)
Comment (damendieta) [reply base_address_parish/models/res_partner.py]: Zip code is different, you can check this here: https://www.codigopostal.gob.ec/

I think they should not be related to the parish code.

File: base_address_parish/README.rst [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
+   :local:
+
+Configuration
+=============
+
+To manage industries, you need to:
+
+* Go to *Contacts > Configuration > Sectors of Activity*.
+
+By default only companies have industries. To activate industries in individuals also,
```
Comment (damendieta) [inline base_address_parish/README.rst]: I don't understand why this is related to Sectors of Activity, this is unrelated.

File: base_address_parish/i18n/es_EC.po [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+#. module: base_address_parish
+#: model:ir.model.fields,field_description:base_address_parish.field_res_parish__city_id
+msgid "City"
+msgstr "Ciudad"
+
+#. module: base_address_parish
+#: model:ir.model.fields,field_description:base_address_parish.field_res_parish__code
+msgid "Code"
+msgstr "Codigo"
```
Comment (damendieta) [inline base_address_parish/i18n/es_EC.po:45]: Código, not Codigo

File: base_address_parish/i18n/base_address_parish.po [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+#. module: base_address_parish
+#: model:ir.model.fields,help:base_address_parish.field_res_country__enforce_parishes
+msgid "Check this box to ensure every address created in that country has a 'Parish' chosen in the list of the city's parishes."
+msgstr ""
+
+
+#. module: base_address_parish
+#: model_terms:ir.actions.act_window,help:base_address_parish.action_res_parish_tree
+msgid "Display and manage the list of all parishes that can be assigned to your partner records. Note that an option can be set on each country separately to enforce any address of it to have a parish in this list."
+msgstr ""
```
Comment (damendieta) [inline base_address_parish/i18n/base_address_parish.po]: Add a newline to get rid of this message, you have this problem in several files.
